### PR TITLE
linter: fixed processing of variables in try-catch

### PR DESCRIPTION
**Pull request by i582**
Now, if a variable is declared in `try` and all `catch`, or `finally`, 
then it will not give a warning that the variable may not be defined.

### CODE REVIEW

File: src/linter/block.go [hunk 1]
```diff
@@ -582,7 +582,9 @@ func (b *blockWalker) withNewContext(action func()) *blockContext {
 }
 
 func (b *blockWalker) handleTry(s *ir.TryStmt) bool {
-	contexts := make([]*blockContext, 0, len(s.Catches)+1)
+	var contexts = make([]*blockContext, 0, len(s.Catches)+1)
```
Comment (quasilyte) [inline src/linter/block.go:585]: Why change `:=` to var?

File: src/tests/checkers/catch_test.go [hunk 2]
```diff
@@ -110,3 +110,111 @@ try {
...
+	test := linttest.NewSuite(t)
+	test.AddFile(`<?php
+function f() {
+	try {
+		$a = 100;
+		$c = 100;
+		$d = 100;
+	} catch (Exception $_) {
+		$a = 200;
+	} catch (ExceptionDerived $_) {
```
Comment (quasilyte) [inline src/tests/checkers/catch_test.go:124]: It seems like we don't give "undefined class" warnings for catch clauses?
This test case should probably define `ExceptionDerived` or do a RunFilterMatch so it's less prone to unrelated changes.

File: src/linter/block.go (hunk 1 above)
Comment (i582) [reply src/linter/block.go:585]: Just so that all variable definitions are the same :)
```go
var contexts = make([]*blockContext, 0, len(s.Catches)+1)
var linksCount int
var finallyCtx *blockContext
```

File: src/tests/checkers/catch_test.go (hunk 2 above)
Comment (i582) [reply src/tests/checkers/catch_test.go:124]: > It seems like we don't give "undefined class" warnings for catch clauses?

It seems so.

File: src/linter/block.go (hunk 1 above)
Comment (quasilyte) [reply src/linter/block.go:585]: `var` is generally used for "zero init".
Please don't use var with assignments anywhere except global scope.

**Review by quasilyte — APPROVED** [review]
